clang/lib/Sema/BoundsAnalysis.cpp

```diff
@@ -103,10 +165,33 @@ void BoundsAnalysis::ComputeGenSets() {
       // Here we have the edges (B1->B2) and (B1->B3). We can add "p:i" only
       // on the true edge. Which means we will add the following entry to
       // Gen[B1]: {B2, p:i}
-      if (!pred->succ_empty() && *pred->succs().begin() == EB->Block)
+
+      // Check if EB is on a true edge of pred. The false edge (including the
+      // default case for a switch) is always the last edge in the list of
+      // edges. So we check whether EB is on the last edge for pred.
+      if (pred->succ_size() &&
+          EB->Block != *(pred->succs().end() - 1)) {
         // Get the edge condition and fill the Gen set.
-        if (Expr *E = GetTerminatorCondition(pred))
+        if (Expr *E = GetTerminatorCondition(pred)) {
+
+          // Check if the pred ends in a switch statement.
```

Comment:
Could you add a TODO about handling the default case better and cite the GitHub issue?

Reply:
Done.